mantidproject/mantid
# Adding FFTPreProcessing into Frequency Domain Analysis

**Do not merge before #20310**

It should be possible to add padding and an apodization function to the data before a FFT. This PR incorporates the FFTPreProcessing algorithm into the Frequency Domain Analysis GUI. The FFT part of the GUI has been tested see #20205 but there has been some refactoring since. 

**To test:**


0. Open muon analysis and load some data (current run will do). **If you do not do this it will not work** 

1. Open Frequency Domain Analysis GUI. 
2. Try different combinations of the advanced options and the spectrum (0) should change. The output will be under the group workspace (e.g. MUSR00023456) 

If you show the hidden workspaces then you can inspect the intermediate outputs. These are the time domain functions multiplied by an equation and with extra zeros added. 

3. If the workspace is set PhaseQuad then the table will look a little different. Before pressing the button make sure the data archive search is on. 

4. This should produce a PhaseTable  and add a workspace to the group (like step 2). 

5. Turning off the generate new phase table and pressing the button does not change the phase table (I suggest opening the phase table first and changing the axis), when the button is pressed it should not change.

6. Turn on the generate new phase table and see that it does change,.   

Fixes #20231 . 

**Release Notes** 

The documentation will be updated in #20312 
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
